DOCSP 43493 VSCode Copilot /query functionality

ltran-mdb2 opened the pull request:
## DESCRIPTION

Adds a page on the MongoDB Github Copilot Participant and the /query command
Adds a page on AI and Data Usage (Based on the[ Compass AI and Data Usage page](https://www.mongodb.com/docs/compass/current/query-with-natural-language/ai-and-data-usage-information/))

## STAGING

https://deploy-preview-92--docs-mongodb-vscode.netlify.app/copilot/
https://deploy-preview-92--docs-mongodb-vscode.netlify.app/ai-data-usage/
https://deploy-preview-92--docs-mongodb-vscode.netlify.app/copilot-query/

## JIRA

https://jira.mongodb.org/browse/DOCSP-43493

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Is this free of spelling errors?
- [x] Is this free of grammatical errors?
- [x] Is this free of staging / rendering issues?
- [x] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

mdb-ashley commented on source/copilot.txt:
```diff
@@ -0,0 +1,77 @@
+.. code-block:: javascript
+   :copyable: false
+
+   {
```
[question] Should we use the edu dataset for examples like this?

ltran-mdb2 replied:
I think it makes sense to do so! I changed the example to use the same Mflix database

mdb-ashley reviewed (approved):
LGTM, just left a question for your consideration. Thanks, Lauren!

GaurabAryal commented on source/ai-data-usage.txt:
```diff
@@ -0,0 +1,36 @@
+---------------------
+
+The MongoDB Github Copilot Participant is powered by `Github Copilot <https://github.com/features/copilot>`__.
+This is subject to change in the future. 
```
I'm not sure if this will ever change since it is an extension for Github Copilot.

ltran-mdb2 replied:
Removed the line about changing!

GaurabAryal commented on source/ai-data-usage.txt:
```diff
@@ -0,0 +1,36 @@
+---------------------
+
+When you use the MongoDB Github Copilot Participant, the following
+information is sent to MongoDB's backend and/or the third party AI
```
The 'third party AI provider' here refer distinct to two things:
1. The LLM API exposed by Copilot itself (we theoretically use the exact same LLM that Copilot uses)
2. Our docs chatbot API endpoint used by our Docs Chatbot

GaurabAryal replied:
Not sure if we need to be that specific, however,

ltran-mdb2 replied:
@themantissa What do you think?

GaurabAryal commented on source/copilot.txt:
```diff
@@ -0,0 +1,80 @@
+|vsce-full| includes a MongoDB Github Copilot Participant to assist in
+using `Github Copilot <https://github.com/features/copilot>`__ with your
+MongoDB deployments. Through Github Copilot's chat feature, users with
+|vsce| can interact with their MongoDB clusters and generate code with
```
Let's also include that they can ask questions about their database collection schema, and ask MongoDB documentation questions that the participant will provide an answer to with specific docs URLs as a reference.

ltran-mdb2 replied:
Done!

GaurabAryal commented on source/copilot.txt:
```diff
@@ -0,0 +1,80 @@
+When the LLM generates a query, you can open the query in a playground
+file or run the query directly in your collection. 
+
+Example
```
Could we provide an example where we can show that users can iteratively build an aggregation with Copilot? Sent over a view of this example to you over slack!

ltran-mdb2 replied:
Done! Because adding this example requires more page space, I made the /query section a separate page so that we can add longer examples for each command without resulting in one super long page.

GaurabAryal reviewed (approved):
LGTM

themantissa commented on source/ai-data-usage.txt:
```diff
@@ -0,0 +1,35 @@
+
+The MongoDB Github Copilot Participant is powered by `Github Copilot <https://github.com/features/copilot>`__.
+
+How Your Data is Used
```
I'd like us to make sure to run this answer by Product legal if we haven't yet. This contains what we normally include but because we get so many questions from customers it's just good to be sure we have covered our bases here.   We've been primarily working with Rachael McClure  - let me know if you need any assistance in getting a check here.  cc @GaurabAryal

themantissa reviewed (changes requested):
One edit and comment responses.   Reach out if any questions.  Thanks!

ltran-mdb2 commented on snooty.toml:
```diff
@@ -40,6 +41,7 @@ cifs = ":abbr:`CIFS (Common Internet File System)`"
 cps = ":abbr:`CPS (Cloud Provider Snapshots)`"
 cmk = ":abbr:`CMK (customer master key)`"
 compass = "MongoDB Compass"
+copilot = "MongoDB Github Copilot Participant"
```
Fixed!

themantissa replied:
Thank you!

ltran-mdb2 commented:
Thanks for your feedback @themantissa ! I added a mention of the docs chatbot and fixed the GitHub spelling error - could you take another look when you get a chance?

themantissa reviewed (approved):
Thank you for the quick turnaround.
